src/common/test/myProfileMocking.ts

```diff
@@ -155,3 +207,189 @@ export const getVerifiedData = (
   },
   ...overrides,
 });
+
+function createNewProfileNode(dataType: EditDataType): MultiItemProfileNode {
+  return {
+    ...formConstants.EMPTY_VALUES[dataType],
+  };
+}
+
+function findNode(
+  nodes: MultiItemProfileNode[],
+  id: string,
+  returnIndex = false
+): number | MultiItemProfileNode | undefined {
+  const func = returnIndex ? 'findIndex' : 'find';
+  return nodes[func](node => node.id === id);
+}
+
+function getEdges(
+  profileData: ProfileData,
+  dataType: EditDataType
+): EdgeList | null {
+  const targetPath = `${dataType}.edges`;
+  return _.get(profileData, targetPath, null);
+}
+
+function getEdgeTypeName(
+  dataType: EditDataType
+): 'EmailNodeEdge' | 'AddressNodeEdge' | 'PhoneNodeEdge' {
+  return dataType === 'emails'
+    ? 'EmailNodeEdge'
+    : dataType === 'phones'
+    ? 'PhoneNodeEdge'
+    : 'AddressNodeEdge';
+}
+
+function getPrimaryPropName(
+  dataType: EditDataType
+): 'primaryAddress' | 'primaryPhone' | 'primaryEmail' {
+  return dataType === 'emails'
+    ? 'primaryEmail'
+    : dataType === 'phones'
+    ? 'primaryPhone'
+    : 'primaryAddress';
+}
+
+function createEdge(
+  dataType: EditDataType,
+  nodeData: Partial<MultiItemProfileNode>
+): Mutable<InsertableEdge> {
+  const typename = getEdgeTypeName(dataType);
+  const node = {
+    ...createNewProfileNode(dataType),
+    ...nodeData,
+  };
+  return {
+    __typename: typename,
+    node,
+  } as InsertableEdge;
+}
+
+function setPrimaryNode(
+  profile: Mutable<ProfileData>,
+  dataType: EditDataType,
+  newPrimary: Mutable<MultiItemProfileNode> | null
+): void {
+  if (newPrimary) {
+    newPrimary.primary = true;
+  }
+  const propName = getPrimaryPropName(dataType);
+  Object.assign(profile, { [propName]: newPrimary });
+}
+
+export function cloneAndManipulateProfile(
+  profileData: ProfileData
+): ManipulationFunctions {
+  const currentProfile = _.cloneDeep(profileData) as Mutable<ProfileData>;
+  // added nodes must have empty id in formValues but id in the created GraphQL
+  // which requires same extra check ups and deletion:
+  const addedNodes: Set<string> = new Set();
+  return {
+    getProfile() {
+      return currentProfile;
+    },
+    add(dataType, data, index = -1) {
```

Comment:
This seems to be missing typings?

Reply:
Not needed, because TS can combine returned object and function's return type (ManipulationFunctions)